src/lua/trigger.c

```diff
@@ -369,15 +369,25 @@ luaT_trigger_list_run(struct lua_State *L)
 	struct rlist *trigger_list = luaT_check_trigger_list(L, 1);
 	int top = lua_gettop(L);
 	struct lbox_trigger *trigger;
-	rlist_foreach_entry(trigger, trigger_list, base.link) {
+	/*
+	 * Firstly save all triggers since they can be deleted. Since we push
+	 * them to the stack, do it in reverse order.
+	 */
+	rlist_foreach_entry_reverse(trigger, trigger_list, base.link) {
```

Comment:
Why not use `trigger_run`, which is safe against trigger deletion?

Reply:
Well, I don't know why I haven't come up with this idea 😄
Done.